### ISSUE TITLE
Change domain name issues

### ISSUE THREAD

**Issue by iliajie** [virtualmin/virtualmin-gpl#472]
Hello, Jamie.

I discovered two issues today that popped upon changing domain name:

Note that all rename tests were run with fully changing a domain name, usernames, and a home dir. All renames were made on the default domain created in the wizard (shouldn't make a different but just mentioning).

---

The first issue happened on Ubuntu 22.04 with Apache, and later I reproduced it on Alma 9 with Nginx, so this issue is web-server independent:

   * If a parent domain is renamed and it has sub domains with FPM pools then any options with a parent path like `upload_tmp_dir` and `session.save_path` are not renamed and as a result FPM service fails to restart.

-----

The second issue only happens on RHEL systems with Apache. Tried on CentOS 7 and Alma 9.

   * Renaming a (default) domain returns an error:
   ```
    Changing username in SSL website configuration ..
    .. Apache SSL website failed! : flush_file_lines called on non-loaded file /etc/httpd/conf/httpd.conf 
   ```
Looking at the `httpd.conf` I can see that only VirtualHost for `http:80` site and `ServerName` was changed, while `ssl:443` version and `ServerName` still had original (before rename) domain name. The paths for SSL certificates in both cases were different, one was a new style `/etc/ssl/virtualmin` and the other was the old style under domain's home directory.

**Comment by jcameron** [comment]
> If a parent domain is renamed and it has sub domains with FPM pools

I assume these PHP options that weren't updated are in the FPM pool config file, not the Apache config?

> Apache SSL website failed! : flush_file_lines called on non-loaded file /etc/httpd/conf/httpd.conf

Does this domain have an `SuexecUserGroup` line in it's Apache config?

**Comment by iliajie** [comment]
> I assume these PHP options that weren't updated are in the FPM pool config file, not the Apache config?

Yes, correct. I was only referring to PHP-FPM config.



> Does this domain have an `SuexecUserGroup` line in it's Apache config?

No, `SuexecUserGroup` is not set and not expected to be set, as these systems only support PHP-FPM.

**Comment by iliajie** [comment]
I ran more tests and also noticed that Virtualmin sub-domain config is not updated with a new path as well! For example `public_html_path` and `awstats_pass` and `cgi_bin_path` and all other, including `user` and `group`.

Also, by looking at sub-server Apache config nothing gets renamed, nor `FCGIWrapper` (if used in FCGId mode) nor `<Directory></Directory>` directives.

**Comment by jcameron** [comment]
Try this patch and see if it helps with one or both of these issues : https://github.com/virtualmin/virtualmin-gpl/commit/253999978c550e55b4c06ac8e1d6dceccb35ae19

**Comment by iliajie** [comment]
Yes, it solves the problem for `flush_file_lines called on non-loaded file` error.

But not for subdomains configs of renamed parent: not for PHP-FPM pools configs, nor for Virtualmin domain config, not any other services configs.

**Comment by jcameron** [comment]
> But not for subdomains configs of renamed parent:

Just to clarify, is the issue that the home directory wasn't updated in the configs of sub-servers?

**Comment by iliajie** [comment]
Yes, in neither configs. Nor Virtualmin domain configs or any other services, like PHP-FPM and Apache.

**Comment by iliajie** [comment]
By the way, not only home directory. Also usernames too.

**Comment by jcameron** [comment]
Looking into this now ..

**Comment by jcameron** [comment]
https://github.com/virtualmin/virtualmin-gpl/commit/3e5ed6ee11d44853c482d7c4f03e99cf481a1300 fixes that sub-server issue

**Comment by iliajie** [comment]
Supreme! Although, one small nagging issue left. Sub-domains PHP-FPM pool files don't get updated with a new `group=` and `listen.group=`. Everything else in the pool files changed correctly, i.e. `user=` and `listen.user` and all paths but not groups:

<img width="601" alt="image" src="https://user-images.githubusercontent.com/4426533/197740053-a883016b-bcb9-4fea-990b-048164afc361.png">

**Comment by jcameron** [comment]
Thanks, I saw this too and I'm still looking into it ...

**Comment by jcameron** [comment]
Ok this has been fixed by https://github.com/virtualmin/virtualmin-gpl/commit/af5d5ed9d53f9b5133a2334d8d71b5bff8f1b0cd

**Comment by iliajie** [comment]
> Ok this has been fixed by [af5d5ed](https://github.com/virtualmin/virtualmin-gpl/commit/af5d5ed9d53f9b5133a2334d8d71b5bff8f1b0cd)

Confirmed it works now. Thanks!
